hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/InsertIntoHoodieTableCommand.scala

```diff
@@ -142,12 +142,26 @@ object InsertIntoHoodieTableCommand extends Logging with ProvidesHoodieConfig wi
     //       positionally for example
     val expectedQueryColumns = catalogTable.tableSchemaWithoutMetaFields.filterNot(f => staticPartitionValues.contains(f.name))
     val coercedQueryOutput = coerceQueryOutputColumns(StructType(expectedQueryColumns), cleanedQuery, catalogTable, conf)
+    val coercedQueryOutputWithoutMetaFields = removeMetaFields(coercedQueryOutput.output)
+    val dataProjectsWithoutMetaFields = getTableFieldsAlias(coercedQueryOutputWithoutMetaFields, StructType(expectedQueryColumns).fields)
     // After potential reshaping validate that the output of the query conforms to the table's schema
     validate(removeMetaFields(coercedQueryOutput.schema), partitionsSpec, catalogTable)
+    val staticPartitionValuesExprs = createStaticPartitionValuesExpressions(staticPartitionValues, targetPartitionSchema)
+    Project(dataProjectsWithoutMetaFields ++ staticPartitionValuesExprs, coercedQueryOutput)
+  }
 
-    val staticPartitionValuesExprs = createStaticPartitionValuesExpressions(staticPartitionValues, targetPartitionSchema, conf)
-
-    Project(coercedQueryOutput.output ++ staticPartitionValuesExprs, coercedQueryOutput)
+  private def getTableFieldsAlias(queryOutputWithoutMetaFields: Seq[Attribute],
+                                  schemaWithoutMetaFields: Seq[StructField]): Seq[Alias] = {
+    queryOutputWithoutMetaFields.zip(schemaWithoutMetaFields).map { case (dataAttr, dataField) =>
+      val targetAttrOption = if (dataAttr.name.startsWith("col")) {
```

Comment:
Why starts with "col"?

Reply:
This is to ignore any unresolved attributes (ideally, we should not hit this situation). This had been there in versions before 0.12.0 - https://github.com/apache/hudi/commit/ea1fbc71ecf9eec20815e2bc51bb8489e4ca1804#diff-1b7609d92cd5c6f871c01031ab923e18470ab5077271639905099241320314e7